ggml.c

```diff
@@ -3550,16 +3550,17 @@ inline static void ggml_vec_log_f32  (const int n, float * y, const float * x) {
 inline static void ggml_vec_abs_f32  (const int n, float * y, const float * x) { for (int i = 0; i < n; ++i) y[i] = fabsf(x[i]); }
 inline static void ggml_vec_sgn_f32  (const int n, float * y, const float * x) { for (int i = 0; i < n; ++i) y[i] = (x[i] > 0.f) ? 1.f : ((x[i] < 0.f) ? -1.f : 0.f); }
 inline static void ggml_vec_step_f32 (const int n, float * y, const float * x) { for (int i = 0; i < n; ++i) y[i] = (x[i] > 0.f) ? 1.f : 0.f; }
-inline static void ggml_vec_tanh_f32 (const int n, float * y, const float * x) { for (int i = 0; i < n; ++i) y[i] = tanhf(x[i]);  }
+inline static void ggml_vec_tanh_f32 (const int n, float * y, const float * x) { for (int i = 0; i < n; ++i) y[i] = tanh(x[i]);  }
```

Comment:
Here, instead of going to double precision (which still does not save you if the values become too big), I would also do
```
inline static float ggml_tanh(float x) {
    return x < -10.f ? -1.f : x < 10.f ? tanh(x) : 1.f;
}
inline static void ggml_vec_tanh_f32(const int n, float * y, const float * x) {
    for (int i = 0; i < n; ++i) y[i] = ggml_tanh(x[i]);
}
```

Reply:
I checked for NaN input here but nothing.